source/postprocess/visualization/heating.cc

```diff
@@ -32,29 +32,59 @@ namespace aspect
     namespace VisualizationPostprocessors
     {
       template <int dim>
-      InternalHeating<dim>::
-      InternalHeating ()
+      Heating<dim>::
+      Heating ()
         :
-        DataPostprocessorScalar<dim> ("Internal heating rate",
-                                      update_values | update_gradients | update_q_points)
+        DataPostprocessor<dim> ()
       {}
 
+      template <int dim>
+      std::vector<std::string>
+      Heating<dim>::
+      get_names () const
+      {
+        std::vector<std::string> names = this->get_heating_model_manager().get_active_heating_model_names();
+        for (unsigned int i=0; i<names.size(); ++i)
+          std::replace(names[i].begin(), names[i].end(), ' ', '_');
```

Comment:
maybe add a comment that we do this replacement to make the names valid names for output variables via DataOut.